AxonDeepSeg/train_network.py

```diff
@@ -66,8 +66,16 @@ def train_model(
     epochs = config["epochs"]
     image_size = config["trainingset_patchsize"]
     thresh_indices = config["thresholds"]
-    checkpoint = config["checkpoint"]
-    checkpoint_period = config["checkpoint_period"]
+    if "checkpoint" in config:
+        checkpoint = config["checkpoint"]
+    else:
+        # For retrocompatibility with old configs
+        checkpoint = None
```

Comment:
Do we require hardcoding of the checkpoint and checkpoint period as they have been defined in the config file already?

Reply:
I harcoded `checkpoint = None` and `checkpoint_period = 5` in case someone want to retrain with an old config where they are not defined. For example, in the config files saved with the default models on OSF.

Reply:
Since you hardcoded already in the config file, that wouldn't change the results. Also, for the code consistency and easy maintainability in the future, I would have removed from the script and kept the parameter values ( as you have already done).

Reply:
Sorry for the delay. I'm not sure I understand your comment. I agree with you that for new config files created with the notebook, the `checkpoint` and `checkpoint_period `will be present and with values `None` and `5`. However, on OSF, the config files saved with our default models don't have those parameters. If we don't hardcode them here, the training will failed if the user load the old config files without adding the new parameters.

Reply:
So, shouldn't we update the model's readme file on the OSF storage to comply with the changes?

Reply:
Yes, that would work to. I honestly don't know what's best. @mathieuboudreau can you take a look? (we can talk about that PR in the meeting tomorrow for some context)

Reply:
@mariehbourget I think we might have covered this at the last meeting, but in case we didn't, here's my thought. Instead of doing patchwork if/elses every time we add a new parameter in the config to preserve retrocompatibility, I think the better strategy would be to write a function that checks the config file as soon as we load it, and stop with an exception if we load a config file that's missing a parameter. Or, possibly even better, we could se the value there and throw a warning to the user that the value was missing, and here's the default value that we set. It would also help centralise the config changes in the future. I think we already have somewhat of a function that checks configs written, but I think it doesn't work perfectly (or as we really need it) so we could maybe build off of that?

I wouldn't edit the README of the models on OSF every time we change the software (e.g. new configs), that'll just become messy and we'll likely forget to do it. Just warn the user when needed from the software side is my view.

Reply:
Thanks @mathieuboudreau  for your valuable suggestion. @mariehbourget , let me know what do you think about it and if in agreement whether you plan to include the function in this PR or would you like to open another PR to resolve this issue, in that case, I will approve this PR.

Reply:
Thanks @vs74 and @mathieuboudreau for your suggestions. As discussed in group meeting, we'll keep the hardcoding for retrocompatibility, as we don't want to do major refactoring at the moment. If we go ahead with the function, it will be in a future PR.